server/src/main/java/org/candlepin/service/impl/DefaultContentAccessCertServiceAdapter.java

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.impl;
+
+import org.candlepin.model.CandlepinQuery;
+import org.candlepin.model.CertificateSerial;
+import org.candlepin.model.CertificateSerialCurator;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Content;
+import org.candlepin.model.ContentAccessCertificate;
+import org.candlepin.model.ContentAccessCertificateCurator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.model.Environment;
+import org.candlepin.model.EnvironmentContent;
+import org.candlepin.model.KeyPairCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerContentCurator;
+import org.candlepin.model.OwnerEnvContentAccess;
+import org.candlepin.model.OwnerEnvContentAccessCurator;
+import org.candlepin.model.Pool;
+import org.candlepin.model.Product;
+import org.candlepin.pki.PKIUtility;
+import org.candlepin.pki.X509ByteExtensionWrapper;
+import org.candlepin.pki.X509ExtensionWrapper;
+import org.candlepin.service.ContentAccessCertServiceAdapter;
+import org.candlepin.util.OIDUtil;
+import org.candlepin.util.Util;
+import org.candlepin.util.X509V3ExtensionUtil;
+
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * DefaultEntitlementCertServiceAdapter
+ */
+public class DefaultContentAccessCertServiceAdapter implements ContentAccessCertServiceAdapter {
```

Comment:
I'd like to see more unity around the interfaces and implementations.  Right now it looks like we have 

```
EntitlementCertServiceAdapter
BaseEntitlementCertServiceAdapter implements EntitlementCertServiceAdapter
ContentAccessCertServiceAdapter completely unrelated to the previous two
DefaultEntitlementCertServiceAdapter extends BaseEntitlementCertServiceAdapter
DefaultContentAccessCertServiceAdapter implements ContentAccessCertServiceAdapter
```

I'd like to see them all in the same family tree at least.  Even better would be if we could move common bits up to a BaseCertServiceAdapter that handles things like writing the JSON blob and basic certificate generation.  Then we have a ContentAccessCertServiceAdapter interface and an EntitlementCertServiceAdapter interface and the implementations both extend the abstract class and implement the appropriate interface.